### PR TITLE
perf: slight optimization on merge states

### PR DESCRIPTION
**Pull request by yzh119**
When cudagraph is enabled, we will still call merge states kernels for short sequence length, which incurs some unnecessary overhead.

This PR accelerates merge states kernel when there is nothing to merge (`num_index_sets=1`).

We can actually write through to the target buffer for small sequence length, but I'm always lazy evaluated and I'll leave it for a future PR (if necessary).

### CODE REVIEW

**Comment by zhyncs** [comment]
The [commit msg](https://github.com/flashinfer-ai/flashinfer/pull/313/commits/90e5473142620a29506454b7df8de2bbbe5093c6) is interesting :P

**Comment by yzh119** [comment]
never mind :)

File: include/flashinfer/attention/cascade.cuh [hunk 1]
```diff
@@ -152,23 +152,40 @@ __global__ void MergeStatesKernel(DTypeIn* __restrict__ V, float* __restrict__ S
   uint32_t tx = threadIdx.x, ty = threadIdx.y;
   uint32_t pos = blockIdx.x;
   uint32_t head_idx = ty;
-  state_t<vec_size> st;
 
-  vec_t<float, vec_size> v_merged_vec;
-  v_merged_vec.fill(0.f);
+  if (num_index_sets > 1) {
```
Comment (Yard1) [inline include/flashinfer/attention/cascade.cuh]: wouldn't it be cleaner to just early return instead of indenting everything?

Comment (yzh119) [reply include/flashinfer/attention/cascade.cuh]: yes you are right, thanks for your suggestion.

**Pull request merged**